controlplane/kubeadm/controllers/helpers.go

```diff
@@ -246,3 +248,87 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 	}
 	return nil
 }
+
+// MachinesNeedingRollout return a list of machines that need to be upgraded.
```

Comment:
It'd be great if in the future we can also attach a reason message to each Machine that needs to be rolled out, so we can print it out and report it

Reply:
We can use conditions for this?

Reply:
Yeah possibly